packages/core/src/utils/resolveConfig.ts

```diff
@@ -1,17 +1,19 @@
-import { PrismConfig, PrismConfigFactory } from '../types';
+import { PartialPrismConfig, PartialPrismConfigFactory } from '../types';
 
-function isPrismConfigFactory<C, I>(val: PrismConfig<C, I>): val is PrismConfigFactory<C, I> {
+function isPrismConfigFactory<C, I>(
+  val: PartialPrismConfig<C, I>
+): val is PartialPrismConfigFactory<C, I> {
   return typeof val === 'function';
 }
 
 export function resolveConfig<Config, Input>(
   input: Input,
-  config: PrismConfig<Config, Input>,
-  defaultConfig?: PrismConfig<Config, Input>
-): Config {
+  config: PartialPrismConfig<Config, Input>,
+  defaultConfig?: PartialPrismConfig<Config, Input>
+): Partial<Config> {
```

Comment:
Would 
```suggestion
): PartialPrismConfig<Config, Input> {
```
work?

Reply:
Nah, the purpose of resolveConfig is to return actual config. *PrismConfig is either actual config or factory.